src/m3ninx/index/types.go

```diff
@@ -83,10 +84,11 @@ type Readable interface {
 // CompiledRegex is a collection of regexp compiled structs to allow
 // amortisation of regexp construction costs.
 type CompiledRegex struct {
-	Simple      *regexp.Regexp
-	FST         *vregex.Regexp
-	PrefixBegin []byte
-	PrefixEnd   []byte
+	Simple       *regexp.Regexp
+	FST          *vregex.Regexp
+	VellumSyntax *syntax.Regexp
```

Comment:
Could you call this `FSTSyntax` to be consistent with the other field which is called `FST`?